uaclient/status.py

```diff
@@ -1,14 +1,53 @@
+import enum
+
+
 class TxtColor:
     OKGREEN = '\033[92m'
     DISABLEGREY = '\033[37m'
     FAIL = '\033[91m'
     ENDC = '\033[0m'
 
 
+@enum.unique
+class ApplicationStatus(enum.Enum):
+    """
+    An enum to represent the current application status of an entitlement
+    """
+    ENABLED = object()
+    DISABLED = object()
+
+    def to_operational_status(self) -> str:
+        mapping = {
+            ApplicationStatus.ENABLED: ACTIVE,
+            ApplicationStatus.DISABLED: INACTIVE,
+        }
+        return mapping[self]
+
+
+@enum.unique
+class ContractStatus(enum.Enum):
+    """
+    An enum to represent whether a user is entitled to an entitlement
+
+    (The value of each member is the string that will be used in status
+    output.)
+    """
+    ENTITLED = 'entitled'
```

Comment:
Further to my above thinking, this was actually something of a compromise; I like the idea of keeping data and presentation separate, whereas this does muddle up that split (because you could index in to ContractStatus using the "presentation" value and get back something meaningful).

The value of `ContractStatus` is only used in one place though (and that's a place we will be keeping, in status output), so I didn't think it was worth the extra code here.